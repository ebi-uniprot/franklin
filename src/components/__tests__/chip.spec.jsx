import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Chip from '../chip';

describe('Chip component', () => {
  it('should render', () => {
    const { asFragment } = render(<Chip>Some content</Chip>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fire onRemove when remove SVG is clicked', () => {
    const onRemove = jest.fn();
    const { getByTestId } = render(
      <Chip onRemove={onRemove}>Some content</Chip>
    );
    const removeIcon = getByTestId('remove-icon');
    fireEvent.click(removeIcon);
    expect(onRemove).toHaveBeenCalled();
  });

  it('should not show remove icon when onRemove prop is not passed', () => {
    const { queryByTestId } = render(<Chip>Some content</Chip>);
    const removeIcon = queryByTestId('remove-icon');
    expect(removeIcon).toBeNull();
  });
});
