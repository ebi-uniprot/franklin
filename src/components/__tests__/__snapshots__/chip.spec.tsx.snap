// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component should render 1`] = `
<DocumentFragment>
  <button
    class="chip"
    disabled=""
    type="button"
  >
    Some content
  </button>
</DocumentFragment>
`;
